test/common.js

```diff
@@ -45,3 +59,39 @@ export function expectAchievementIdsNested(promise, name,
 			.property(`${name}.titleId`, titleId)
 	]);
 }
+
+export function generateProcessEdit(func, name, rev) {
+	return () => testData.createEditor()
+		.then((editor) => Achievement.processEdit(orm, editor.id))
+		.then((edit) => edit[func][`${name} ${rev}`]);
+}
+
+export function generateProcessEditNamed(func, name) {
+	return () => new Editor({name: testData.editorAttribs.name})
+		.fetch()
+		.then((editor) =>
+			Achievement.processEdit(orm, editor.id)
+		)
+		.then((edit) => edit[func][name]);
+}
+
+export function rewireAchievement(rewiring) {
+	return () => Achievement.__set__(rewiring);
```

Comment:
This shouldn't return a function. Also, better to not use globals, so I'd say pass in the `Achievement` object and then change it within the function (and go back to calling `const Achievement = rewire(...)` in each Achievement module.

There may be unintended side effects to having a shared Achievement variable (not being cleaned up properly after each test, for example).

Reply:
What about changing this to use the `__with__` method?